src/Components/Promises/PromiseTableAdminPanel/index.js

```diff
@@ -0,0 +1,80 @@
+/* @flow */
+import React from 'react';
+import SubjectSelector from '../../Subjects/SubjectSelector';
+import RaisedButton from 'material-ui/RaisedButton';
+import type { SubjectId } from '../../../types';
+import { createSubject } from '../../../utils/api';
+
+const styles = {
+  container: {
+    display: 'flex',
+  },
+  select: {
+    flexGrow: 1,
+  },
+  buttonContainer: {},
+};
+
+type Props = {
+  linkSubjectsCallback: (subjectId: SubjectId) => void,
+};
+
+type State = {
+  subject: ?string | SubjectId,
+};
+
+type ReactSelectValue = {
+  value: ?string | number,
+};
+
+export class PromiseTableAdminPanel extends React.Component<void, Props, State> {
+  props: Props;
+  state = {
+    subject: null,
+  };
+
+  updateSubject = (selectedValue: ?ReactSelectValue) => {
+    const subject = selectedValue && selectedValue.value;
+
+    this.setState({ subject: subject });
+  };
+
+  onLinkSubmit = async (subject: string | SubjectId) => {
+    const { linkSubjectsCallback } = this.props;
+    let subjectId = subject;
+    if (typeof subjectId === 'string') {
```

Comment:
Yeah this is pretty spaghetti.

The idea is that `PromiseTableAdmin` is responsible for returning a SubjectId, whether the subject exists on the backend currently or not.
_This can be seen by the propType `linkSubjectsCallback`_

`react-select` however, can return 
- null (nothing selected)
- `{value: number}` (existing Subject selected)
- `string` user created subject